core/trino-main/src/main/java/io/trino/sql/planner/LogicalPlanner.java

```diff
@@ -624,7 +633,11 @@ private RelationPlan createTableWriterPlan(
                 // empty connector partitioning handle means evenly partitioning on partitioning columns
```

Comment:
In `if (partitioningHandle.isPresent()) {` assert that `getMaxWriterTasks(session)` is empty. You could additionally add comment to SPI `getMaxWriterTasks`